src/crypto/openssl/cose_sign.cpp

```diff
@@ -216,8 +217,16 @@ namespace ccf::crypto
         fmt::format("Can't finish QCBOR encoding with error code {}", err));
     }
 
-    return {
-      static_cast<const uint8_t*>(signed_cose.ptr),
-      static_cast<const uint8_t*>(signed_cose.ptr) + signed_cose.len};
+    if (signed_cose.ptr != underlaying_buffer.data())
+    {
+      LOG_FAIL_FMT("COSE sign1: QCBOR buffer doesn't match underlaing buffer");
```

Comment:
```suggestion
      LOG_FAIL_FMT("COSE sign1: QCBOR buffer doesn't match underlying buffer");
```

Reply:
That does not seem like something we ought to check at runtime, it's either a guarantee the library offers (and which we can confirm in unit tests if necessary), or it's not and we should not rely on it.